qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,333 @@
+            If a constraint has a float coefficient,
+            return the default value for the penalty factor.
+        """
+
+        default_penalty = 1e5
+
+        # Check coefficients of constraints.
+        # If a constraint has a float coefficient, return the default value for the penalty factor.
+        terms = []
+        for constraint in problem.linear_constraints:
+            terms.append(constraint.rhs)
+            terms.extend(coef for coef in constraint.linear.to_dict().values())
+        if any(isinstance(term, float) and not term.is_integer() for term in terms):
+            logger.warning(
+                "Warning: Using %f for the penalty coefficient because "
+                "a float coefficient exists in constraints. \n"
+                "The value could be too small. "
+                "If so, set the penalty coefficient manually.",
+                default_penalty,
+            )
+            return default_penalty
+
+        # (upper bound - lower bound) can be calculate as the sum of absolute value of coefficients
+        # Firstly, add 1 to guarantee that infeasible answers will be greater than upper bound.
+        penalties = [1.0]
+        # add linear terms of the object function.
+        penalties.extend(abs(coef) for coef in problem.objective.linear.to_dict().values())
+        # add quadratic terms of the object function.
+        penalties.extend(abs(coef) for coef in problem.objective.quadratic.to_dict().values())
+
+        return fsum(penalties)
+
+    def interpret(self, x: Union[np.ndarray, List[float]]) -> np.ndarray:
+        """Convert the result of the converted problem back to that of the original problem
+
+        Args:
+            x: The result of the converted problem or the given result in case of FAILURE.
+
+        Returns:
+            The result of the original problem.
+
+        Raises:
+            QiskitOptimizationError: if the number of variables in the result differs from
+                                     that of the original problem.
+        """
+
+        if len(x) != self._src_num_vars:
+            raise QiskitOptimizationError(
+                "The number of variables in the passed result differs from "
+                "that of the original problem."
+            )
+        return np.asarray(x)
+
+    @property
+    def penalty(self) -> Optional[float]:
+        """Returns the penalty factor used in conversion.
+
+        Returns:
+            The penalty factor used in conversion.
+        """
+
+        return self._penalty
+
+    @penalty.setter
+    def penalty(self, penalty: Optional[float]) -> None:
+        """Set a new penalty factor.
+
+        Args:
+            penalty: The new penalty factor.
+                     If None is passed, a penalty factor will be automatically calculated
+                     on every conversion.
+        """
+
+        self._penalty = penalty
+        self._should_define_penalty = penalty is None  # type: bool
```

Comment:
You don't need `_should_define_penalty`. You can do the same thing just checking whether `self._penalty is None` or not.

Reply:
`_should_define_penalty` was introduced in `LinearEqualityToPenalty` to return the correct penalty even it's reused (see issue https://github.com/Qiskit/qiskit-optimization/issues/34 for more details). I think we can have `_should_define_penalty` in `LinearInequalityToPenalty` as well.

Reply:
Thanks. I missed the reuse case. Btw, when I simplified converters, I removed `_should_define_penalty` of `LinearEqualityToPenalty` #174. I revert it asap.